### PR TITLE
feat: support Hive import and export

### PR DESCRIPTION
**Pull request by vagetablechicken**
```
load data infile 'hive://'
select into outfile 'hive://'
```
Options: only mode works, ignore other options(format/delimiter/...). And soft-copy from hive is unsupported now.

### CODE REVIEW

File: java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/utils/HybridseUtil.scala [hunk 1]
```diff
@@ -293,22 +292,21 @@ object HybridseUtil {
...
+    var df = if (format != "csv") {
       // When reading Parquet files, all columns are automatically converted to be nullable for compatibility reasons.
       // ref https://spark.apache.org/docs/3.2.1/sql-data-sources-parquet.html
-      df = reader.format(format).load(file)
+      val df = reader.format(format).load(file)
       require(checkSchemaIgnoreNullable(df.schema, oriSchema),
         s"schema mismatch(ignore nullable), loaded ${df.schema}!= table $oriSchema, check $file")
       // reset nullable
-      df = df.sqlContext.createDataFrame(df.rdd, oriSchema)
+      df.sqlContext.createDataFrame(df.rdd, oriSchema)
```
Comment (vagetablechicken) [inline java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/utils/HybridseUtil.scala:303]: test, data null, schema not null, reset nullable ok?

File: java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/utils/HybridseUtil.scala [hunk 2]
```diff
@@ -321,15 +319,37 @@ object HybridseUtil {
...
+    }
+    val df = spark.sql(s"SELECT * FROM ${file.substring(deli + 1)}")
+
+    val (oriSchema, readSchema, tsCols) = HybridseUtil.extractOriginAndReadSchema(columns)
     if (logger.isDebugEnabled()) {
-      logger.debug("read dataframe count: {}", df.count())
+      logger.debug(s"read dataframe schema: ${df.schema}, count: ${df.count()}")
       df.show(10)
     }
+    require(checkSchemaIgnoreNullable(df.schema, oriSchema), //df.schema == oriSchema, hive table always nullable?
```
Comment (vagetablechicken) [inline java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/utils/HybridseUtil.scala:353]: dataframe schema needs reset?

File: java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/utils/HybridseUtil.scala [hunk 3]
```diff
@@ -321,15 +319,37 @@ object HybridseUtil {
...
+      logger.debug(s"read dataframe schema: ${df.schema}, count: ${df.count()}")
+      df.show(10)
+    }
     require(df.schema == oriSchema, s"schema mismatch, loaded ${df.schema} != table $oriSchema, check $file")
+    df
+  }
+
+  def hiveLoad(spark: SparkSession, file: String, columns: util.List[Common.ColumnDesc]): DataFrame = {
+    // only support thrift No effects
+    require(file.startsWith("hive://"))
```
Comment (vagetablechicken) [inline java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/utils/HybridseUtil.scala]: file tolowercase ,HDFS/HIVE

File: docs/zh/reference/sql/dml/LOAD_DATA_STATEMENT.md [hunk 4]
```diff
@@ -95,3 +98,40 @@ set @@execute_mode='offline';
...
+## Hive 支持
+
+为了支持读Hive，我们需要Hive相关的依赖和Hive配置。
+
+### Spark Hive依赖
+
+我们需要在Spark中编译出Hive依赖，依赖包在`assembly/target/scala-xx/jars`. 将所有依赖加入Spark的class path中。
+
+```
+./build/mvn -Pyarn -Phive -Phive-thriftserver -DskipTests clean package
```
Comment (vagetablechicken) [inline docs/zh/reference/sql/dml/LOAD_DATA_STATEMENT.md:110]: Spark repo cicd add hive

**Review by vagetablechicken — COMMENTED** [review]
need test

File: docs/zh/reference/sql/dml/LOAD_DATA_STATEMENT.md [hunk 5]
```diff
@@ -95,3 +98,40 @@ set @@execute_mode='offline';
...
+```
+
+### Hive 配置
+
+我们推荐使用metastore服务来连接Hive。
```
Comment (vagetablechicken) [inline docs/zh/reference/sql/dml/LOAD_DATA_STATEMENT.md]: 目前只支持

File: docs/zh/reference/sql/dml/LOAD_DATA_STATEMENT.md (hunk 4 above)
Comment (vagetablechicken) [reply docs/zh/reference/sql/dml/LOAD_DATA_STATEMENT.md:110]: see https://github.com/4paradigm/spark/pull/37

**Review by dl239 — APPROVED** [review]

**Review by lqy222 — APPROVED** [review]

File: java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/utils/HybridseUtil.scala (hunk 1 above)
Comment (vagetablechicken) [reply java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/utils/HybridseUtil.scala:303]: reset won't check the data, just modify the schema. The nullable property will affect when we read the real data(e.g. do select).
Note that, the stored data in openmldb will load by autoLoad too, so the schema will be reset.
So the process is: read from hive, hive schema, save to offline storage(parquet), register by autoLoad(reset the schema).
If we reset the schema in hiveLoad, will we get the error when save to offline storage, more early?
online import from hive will report the error if schema not null but the data is null. All we need to do is checking the offline import.

**Pull request merged**